PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx

```diff
@@ -0,0 +1,435 @@
+      if (candidate.flagMcMatchRec() != decayFlag) {
+        continue;
+      }
+
+      selectedKine = false;
+      selectedPid = false;
+      mass = candidate.invMassCharmBaryon();
+      pt = RecoDecay::sqrtSumOfSquares(candidate.pxCharmBaryon(), candidate.pyCharmBaryon());
+
+      // all candidates (candidateCreator)
+      hCandidates->Fill(kHFStepTracked, pt, mass, candidate.collisionMatched(), candidate.originRec());
+
+      // check xi-pi candidate passed candidate selector cuts (except PID)
+      if (candidate.resultSelections() && candidate.statusInvMassLambda() && candidate.statusInvMassCascade() && candidate.statusInvMassCharmBaryon()) {
+        hCandidates->Fill(kHFStepTrackedCuts, pt, mass, candidate.collisionMatched(), candidate.originRec());
+        selectedKine = true;
+      }
+      if (!selectedKine) {
+        continue;
+      }
+
+      // check xi-pi candidate passed candidate selector cuts (PID included)
+      if (candidate.statusPidCharmBaryon()) {
+        hCandidates->Fill(kHFStepTrackedSelected, pt, mass, candidate.collisionMatched(), candidate.originRec());
+        selectedPid = true;
+      }
+      if (!selectedPid) {
+        continue;
+      }
+    }
+  }
+
+  // candidates -> join candidateCreator, candidateCreator McRec and candidateSelector tables
+  // genParticles -> join aod::McParticles and candidateCreator McGen tables
+  template <typename T1, typename T2>
+  void candidateFullLoop(T1& candidates, T2& genParticles, TracksWithSelectionMC const& tracks, aod::McCollisionLabels const& colls, int pdgCode)
+  {
+    // fill hCandidates histogram
+    candidateLoop(candidates, pdgCode);
+
+    auto hCandidates = registry.get<StepTHn>(HIST("hCandidates"));
+    auto hTrackablePtEta = registry.get<StepTHn>(HIST("hTrackablePtEta"));
+
+    // lists for optimization
+    std::vector<bool> tracked(genParticles.size(), false);
+    std::vector<bool> hasITS(genParticles.size(), false);
+    std::vector<bool> hasTPC(genParticles.size(), false);
+    std::vector<bool> selectedIts(genParticles.size(), false);
+    std::vector<bool> selectedItsTpc(genParticles.size(), false);
+    for (const auto& track : tracks) {
+      if (track.mcParticleId() >= 0) {
+        tracked[track.mcParticleId()] = true;
+        if (track.hasITS()) {
+          hasITS[track.mcParticleId()] = true;
+        }
+        if (track.hasTPC()) {
+          hasTPC[track.mcParticleId()] = true;
+        }
+        if (checkTrackGlbTrk(track)) {
+          selectedItsTpc[track.mcParticleId()] = true;
+        }
+        if (checkTrackItsTrk(track)) {
+          selectedIts[track.mcParticleId()] = true;
+        }
+      }
+    }
+
+    auto mass = pdg->Mass(pdgCode);
+
+    int decayFlagGen = 0;
+    if (pdgCode == Pdg::kXiC0) {
+      decayFlagGen = 4;
+    } else if (pdgCode == Pdg::kOmegaC0) {
+      decayFlagGen = 2;
+    } else {
+      LOGP(fatal, "Not implemented for PDG code: ", pdgCode);
+    }
```

Comment:
Use `enum` values.

Reply:
Could you please clarify how you would implement this?
This function is called only with two possible pdgCodes (depending on the process function, see processXic0 and processOmegac0) and there are no other alternatives, moreover there will be no extension of the possible "cases"

Reply:
You compare the value of `decayFlagGen` with the value of the column `hf_cand_toxipi::FlagMcMatchGen`, which is filled with `BIT(OmegaczeroToXiPi)` or `BIT(XiczeroToXiPi)`. So you should use those values for `decayFlagGen` instead of explicit `4` and `2` which might stop working if the `enum` change.